diff --git a/src/dvd_input.c b/src/dvd_input.c
index d7dad87..68c3986 100644
--- a/src/dvd_input.c
+++ b/src/dvd_input.c
@@ -23,13 +23,15 @@
 #include <stdio.h>                               /* fprintf */
 #include <stdlib.h>                              /* free */
 #include <fcntl.h>                               /* open */
-#include <unistd.h>                              /* lseek */
 #include <string.h>                  /* strerror */
 #include <errno.h>
 
 #ifdef _WIN32
 #include <windows.h>
+#include <io.h>
 #include "../msvc/contrib/win32_cs.h"
+#else
+#include <unistd.h>                              /* lseek */
 #endif
 
 #include "dvdread/dvd_reader.h"      /* DVD_VIDEO_LB_LEN */
diff --git a/src/dvdread_internal.h b/src/dvdread_internal.h
index bf4e2e1..c588abc 100644
--- a/src/dvdread_internal.h
+++ b/src/dvdread_internal.h
@@ -24,6 +24,8 @@
 #include <sys/types.h>
 
 #ifdef _WIN32
+#include <windows.h>
+#else
 # include <unistd.h>
 #endif /* _WIN32 */
 
