--- a/configure
+++ b/configure
@@ -7125,10 +7122,11 @@ check_optflags -fno-math-errno
 check_optflags -fno-signed-zeros
 
 if enabled lto; then
-    test "$cc_type" != "$ld_type" && die "LTO requires same compiler and linker"
-    check_cflags  -flto
-    check_ldflags -flto $cpuflags
     disable inline_asm_direct_symbol_refs
+    # Clang's LTO fails on Windows, when there are references outside
+    # of inline assembly to nonlocal labels defined within inline assembly,
+    # see https://github.com/llvm/llvm-project/issues/76046.
+    disable inline_asm_nonlocal_labels
 fi
 
 enabled ftrapv && check_cflags -ftrapv
